### DISCUSSION TITLE
Context separation and REPL

### DISCUSSION THREAD

**Discussion by IngoVals** [counterfact/api-simulator#819] [Q&A] [answered]
According to docs

> The context object is defined in _.context.ts in the same directory as the file that uses it. It's up to you to define the API for a context object. For example, your _.context.ts file might look like this.
> By default, each _.context.ts delegates to its parent directory, so you can define one context object in the root _.context.ts and use it everywhere.

So we could have this separation if wanted. for example here

```
.
└── counterfact/
    ├── components
    ├── path-types
    └── paths/
        ├── _context.ts
        └── api/
            ├── person/
            │   ├── _context.ts
            │   ├── getPerson.ts
            │   ├── updatePerson.ts
            │   └── deletePerson.ts
            ├── course
            └── teacher
```

When working with `getPerson`, `updatePerson `etc. it should use the inner `_context `file right?

How would I access this inner context file through the REPL ?

**Comment by IngoVals** [comment]
There is a `loadContext` command with in the REPL I just found.

But I'm having issues with functions defined in the inner context not getting loaded.

**Reply by pmcelhaney** [reply]
Sorry, I haven't been paying attention to discussions and just saw this! Are you still having issues?

**Reply by pmcelhaney** [reply]
It's actually not working for either then main context or the inner context. But I think that was because of a change that was made more recently than this post. Fix is forthcoming. 

#898

**Reply by pmcelhaney** [reply]
That fix has been released now.

**Comment by dethell** [comment, marked as answer]
You are correct that `loadContext` is the key here. In your example, if your `api/person/_context.ts` file has implementations that `getPerson.ts` uses, then you could load them like this:
```
const personContext = loadContext('/api/person`);
personContext.getPerson('123'); // This will echo the results of that function to the console.
```

**Reply by IngoVals** [reply]
ah, now I get it. I can loadContext into a property on the main context, or perhaps even just live in the REPL. Brilliant.

**Reply by pmcelhaney** [reply]
There's currently not a way to call `loadContext()` from the code. It's only available in the REPL. Should I investigate making it available from the code?
